packages/deployer/internal/process-contracts.js

```diff
@@ -39,11 +37,6 @@ async function initContractData(contractFullyQualifiedName, extraData = {}) {
 async function _initContractSource(contractFullyQualifiedName) {
   const { deployment } = hre.deployer;
 
-  // If the contract sources are already calculated, don't re do it.
-  if (deployment.sources[contractFullyQualifiedName]) {
-    return;
-  }
```

Comment:
This check was working as we save the sources using `sourceName`, not `fullyQualifiedNames` now